pkg/dao/resource_component.go

```diff
@@ -116,3 +118,36 @@ func ResourceComponentGetUniqueKey(r *model.ResourceComponent) string {
 	// Align to schema definition.
 	return strs.Join("-", string(r.ConnectorID), r.Shape, r.Mode, r.Type, r.Name)
 }
+
+// clearComponents clears the components that not exist in the database.
+func clearComponents(
+	ctx context.Context,
+	mc model.ClientSet,
+	components model.ResourceComponents,
+) (model.ResourceComponents, error) {
```

Comment:
i didn't see any delete/remove action in this function, can we clarify the purpose of this function?

Reply:
updated, we need refactor the resource components update logic in 0.6.0.